structure/base/base.go

```diff
@@ -116,3 +120,20 @@ func (b *Base) WithReference(reference string) *Base {
 	}
 	return base
 }
+
+func (b *Base) WithReferenceNoWarning(reference string) *Base {
+	base := &Base{
```

Comment:
I know the rest of the code do the same, but:
```suggestion
	base := b.WithReference(reference)
```
without the duplicate code might be better ?

Reply:
this code is removed ;-)